add documentation for manual job trigger (#939) (#965)
kyma-project/test-infra: docs/prow/prow-jobs.md

- [Create component jobs](./component-jobs.md)
- [Create release jobs](./release-jobs.md)

For further reference, read a more technical insight into the Kubernetes [ProwJobs](https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md).

> ## Manual job trigger

Reviewer: ```suggestion
## Trigger jobs manually
```

> If you need to trigger a new job, because re-running the job would would not use the new code base, you can do so manually.

Reviewer: ```suggestion
Re-running the job does not always mean using the new code base. In such a case, you can trigger the job manually.
```

I have a question to this one - am I understanding it correctly - does re-running a job always mean that the new code base is not used? Or just in some cases? If it's always the case I will rephrase this sentence :)

Author: From the current Prow page, you can get a `kubectl` command to re-run the selected job. Which means it uses the exact same commit it was running with before. If you updated something, it would not run the new code/config.
Example is maybe a job that only runs every 4 hours. To not wait for the next execution that would be 4 hours from now, you could run it manually, to verify it works.

Reviewer: I updated the suggestion :)

> For that you'll need a tool from the called [mkpj](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/mkpj) that will generate a valid ProwJob yaml for you. In the example below `kyma-gke-nightly` is target to be triggered manually:

Reviewer: ```suggestion
Use a tool called [mkpj](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/mkpj) that generates a valid `yaml` for the ProwJob. See the example of generating the `kyma-gke-nightly` target:
```

```shell
go run prow/cmd/mkpj/main.go --job=kyma-gke-nightly --config-path="$GOPATH/src/github.com/kyma-project/test-infra/prow/config.yaml" --job-config-path="$GOPATH/src/github.com/kyma-project/test-infra/prow/jobs/" > job.yaml
```

> The job can then be triggered with `kubectl`:

Reviewer: ```suggestion
You can trigger this job manually. Run:
```
